### ISSUE TITLE
Image gaps vs refetching too much data

### ISSUE THREAD

**Issue by dcarosone** [dcarosone/scrape-cassini#2]
Gaps in the image sequence will be retried every run.

It needs to be smarter about:
- recording the known gaps
- determining whether the gap is internal, or at the top of the range (we've run off into images that don't exist yet).

See #1.

**Comment by dcarosone** [comment]
With the new website and JSON-based data (#5), this no-longer happens, because we get image metadata in page-size chunks and only process what gets listed.

The issue has morphed a little; instead we fetch a bunch of metadata pages on each run, and will often get repeat data on images we already know about.  Because the page numbers are relative to the most recent image, it's hard to work around this trivially.  Gaps in our data could be:
- missing images
- runs that terminated early and didn't get back as far as what we previously knew
- infrequent runs, that didn't look back as far as what we previously knew

For now, the simple-minded approach is just to re-fetch too much metadata to be sure and cover the gap.

**Comment by dcarosone** [comment]
Solution in two steps:
1. Provide an option to specify how many pages to fetch, and some more meaningful messages (part of #6) to help spot whether you're getting too many / enough / too few and have gaps.
2. Be smarter; perhaps something like:
   - fetch pages in reverse order (images forward in time).
   - guess how many pages, using the total counts and id numbers / 48 from a probe page. Step backwards extra pages if we hit gaps.
   - warn of gaps larger than a page worth?

**Comment by dcarosone** [comment]
If we fetch pages forward in time (decreasing page number) there is a race.  Images may be added at the top of the list, pushing others onto the next page, and we can miss them.

If we fetch pages backward in time (as now), it's hard to know how far back to go.  It's not enough to go back to the point where we find known images, because previous scans went backwards too, and may not have gone as far back as needed (terminated early).  There might also be a race as above, images we just found in the previous page have now been pushed to the next, and we decide to stop too soon because we found known images.

Using the total counts and other api pages as additional clues could help, but needs this state to be kept.  Which means we'll probably end up keeping two lots of state; the current TSV and a fuller JSON version similar to the api return.

An option may be to run as a persistent service and poll the totals and the first page semi-regularly.

Meanwhile, the advice in the readme works well enough for step 1, above.
